pytorch_lightning/trainer/connectors/model_connector.py

```diff
@@ -55,6 +46,11 @@ def copy_trainer_model_properties(self, model):
             m.local_rank = self.trainer.local_rank
 
     def get_model(self):
-        is_dp_module = isinstance(self.trainer.model, (LightningDistributedDataParallel, LightningDataParallel))
-        model = self.trainer.model.module if is_dp_module else self.trainer.model
-        return model
+        return self._reference_model(self.trainer.model)
+
+    def _reference_model(self, model):
+        if self.trainer.accelerator_backend:
+            ref_model = self.trainer.accelerator_backend.reference_model(model)
+        else:
+            ref_model = model
+        return ref_model
```

Comment:
I'm not the biggest fan of having to deal with the model not being wrapped within the accelerator. I'm thinking of changing this to:

```python
if isinstance(model, LightningModule):
    return model
elif self.trainer.accelerator_backend:
    ref_model = self.trainer.accelerator_backend.reference_model(model)
else:
    raise RuntimeError("Model reference could not be found")
```

But I'm not sure what the edge cases are for this function.

Reply:
Going back and forth here but I think on par I'd prefer allowing the accelerator to deal with all logic here, this allows for more extensibility in the future within the accelerator and less prone to errors.